strawberry/experimental/pydantic/v2_compat.py

```diff
@@ -0,0 +1,104 @@
+import dataclasses
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, Type
+
+import pydantic
+from pydantic import BaseModel
+from pydantic.version import VERSION as PYDANTIC_VERSION
+
+if TYPE_CHECKING:
+    from pydantic.fields import FieldInfo
+
+IS_PYDANTIC_V2: bool = PYDANTIC_VERSION.startswith("2.")
+IS_PYDANTIC_V1: bool = not IS_PYDANTIC_V2
+
+
+@dataclass
+class CompatModelField:
+    name: str
+    type_: Any
+    outer_type_: Any
+    default: Any
+    default_factory: Optional[Callable[[], Any]]
+    required: bool
+    alias: Optional[str]
+    allow_none: bool
+    has_alias: bool
+    description: Optional[str]
+
```

Comment:
we have a compat file (inspired by how fastapi does it) to work with both versions